core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -174,8 +174,9 @@ class DockerContainer(protected val id: ContainerId,
   override def suspend()(implicit transid: TransactionId): Future[Unit] = {
     super.suspend().flatMap(_ => if (useRunc) runc.pause(id) else docker.pause(id))
   }
-  def resume()(implicit transid: TransactionId): Future[Unit] =
-    if (useRunc) { runc.resume(id) } else { docker.unpause(id) }
+  override def resume()(implicit transid: TransactionId): Future[Unit] = {
+    if (useRunc) { runc.resume(id) } else { docker.unpause(id) }.map(_ => super.resume())
```

Comment:
flatMap

Reply:
when in doubt refactor...

Reply:
good question @sven-lange-last. Better safe than sorry and make an interim value or put parentheses around the if expression.

Reply:
I don't think it's needed, but agree it will at least be more clear. Also changed to flatMap